bin/validate_gradle_wrapper

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash -eu
+
+CHECKSUM_URLS=$(curl -s https://services.gradle.org/versions/all | jq -r ".[].wrapperChecksumUrl? | select(.)")
+
+function validate_checksum() {
+    wrapper_file="$1"
+    validated_with_checksum_from_url=""
+
+    wrapper_checksum=$(sha256sum "$wrapper_file")
+    # Remove the file location from the "wrapper_checksum" result
+    sha256_checksum_to_be_validated="${wrapper_checksum%% *}"
+
+    while IFS= read -r checksum_url; do
+        downloaded_checksum=$(curl -s --location "$checksum_url")
+        if [[ "$sha256_checksum_to_be_validated" == "$downloaded_checksum" ]]; then
+            validated_with_checksum_from_url="$checksum_url"
+            break;
+        fi
+    done <<< "$CHECKSUM_URLS"
+
+    if [[ -z "$validated_with_checksum_from_url" ]]; then
+        echo "Failed to validate '$wrapper_file'"
+        exit 1
+    else
+        echo "'$wrapper_file' is validated with sha256 checksum from '$validated_with_checksum_from_url'"
```

Comment:
💄  small nit (to not make it look like an "in-progress" message but more like a "done" message):
```suggestion
        echo "'$wrapper_file' was validated with sha256 checksum from '$validated_with_checksum_from_url'"
```

Reply:
An in-progress message would be `is being validated` and I believe, because the log is created at the time of the validation, the current tense is correct.